<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="windowModality">
   <enum>Qt::WindowModality::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>714</width>
    <height>592</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>UIAP</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resource.qrc">
    <normaloff>:/icon/imgs/firmware_update.png</normaloff>:/icon/imgs/firmware_update.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_9">
     <item>
      <widget class="QGroupBox" name="serial_cfg_group_box">
       <property name="title">
        <string>串口配置</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="2" column="1" colspan="2">
         <widget class="QComboBox" name="serial_data_bits_cmb"/>
        </item>
        <item row="1" column="1" colspan="2">
         <widget class="QComboBox" name="serial_baud_rate_cmb">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="serial_port_lb">
          <property name="text">
           <string>端口号</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1" colspan="2">
         <widget class="QComboBox" name="serial_stop_bits_cmb"/>
        </item>
        <item row="4" column="1" colspan="2">
         <widget class="QComboBox" name="serial_parity_cmb"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="serial_baud_rate_lb">
          <property name="text">
           <string>波特率</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="serial_data_bits_lb">
          <property name="text">
           <string>数据位</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1" colspan="2">
         <widget class="QComboBox" name="serial_port_cmb"/>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="serial_stop_bits_lb">
          <property name="text">
           <string>停止位</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="serial_flow_ctrl_lb">
          <property name="text">
           <string>流控制</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="serial_parity_lb">
          <property name="text">
           <string>校验位</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1" colspan="2">
         <widget class="QComboBox" name="serial_flow_ctrl_cmb"/>
        </item>
        <item row="6" column="0">
         <widget class="QCheckBox" name="serial_rts_chk">
          <property name="text">
           <string>RTS</string>
          </property>
         </widget>
        </item>
        <item row="8" column="0" colspan="3">
         <widget class="QPushButton" name="serial_open_btn">
          <property name="text">
           <string>打开串口</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QCheckBox" name="serial_dtr_chk">
          <property name="text">
           <string>DTR</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="firmware_cfg_group_box">
       <property name="title">
        <string>固件配置</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_3" columnstretch="9,0,1">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_2" columnstretch="0,9,0,0,1,0">
          <item row="3" column="0">
           <widget class="QCheckBox" name="firmware_selected_chk_2">
            <property name="text">
             <string>#2</string>
            </property>
           </widget>
          </item>
          <item row="0" column="5">
           <widget class="QLabel" name="firmware_crypto_lb">
            <property name="text">
             <string>加密算法</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QCheckBox" name="firmware_selected_chk_5">
            <property name="text">
             <string>#5</string>
            </property>
           </widget>
          </item>
          <item row="3" column="5">
           <widget class="QComboBox" name="firmware_crypto_cmb_2"/>
          </item>
          <item row="9" column="3">
           <widget class="QToolButton" name="firmware_open_file_tb_7">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QLabel" name="firmware_address_lb">
            <property name="text">
             <string>地址</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1" colspan="2">
           <widget class="QLineEdit" name="firmware_path_le_3"/>
          </item>
          <item row="7" column="4">
           <widget class="QLineEdit" name="firmware_address_le_6"/>
          </item>
          <item row="2" column="5">
           <widget class="QComboBox" name="firmware_crypto_cmb_1"/>
          </item>
          <item row="6" column="3">
           <widget class="QToolButton" name="firmware_open_file_tb_5">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="5" column="5">
           <widget class="QComboBox" name="firmware_crypto_cmb_4"/>
          </item>
          <item row="5" column="4">
           <widget class="QLineEdit" name="firmware_address_le_4"/>
          </item>
          <item row="9" column="4">
           <widget class="QLineEdit" name="firmware_address_le_7"/>
          </item>
          <item row="9" column="5">
           <widget class="QComboBox" name="firmware_crypto_cmb_6"/>
          </item>
          <item row="5" column="3">
           <widget class="QToolButton" name="firmware_open_file_tb_4">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="3">
           <widget class="QLabel" name="firmware_path_lb">
            <property name="text">
             <string>路径</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QCheckBox" name="firmware_selected_chk_1">
            <property name="text">
             <string>#1</string>
            </property>
           </widget>
          </item>
          <item row="6" column="4">
           <widget class="QLineEdit" name="firmware_address_le_5"/>
          </item>
          <item row="1" column="0">
           <widget class="QCheckBox" name="firmware_selected_chk_0">
            <property name="text">
             <string>#0</string>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="7" column="5">
           <widget class="QComboBox" name="firmware_crypto_cmb_7"/>
          </item>
          <item row="4" column="3">
           <widget class="QToolButton" name="firmware_open_file_tb_3">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1" colspan="2">
           <widget class="QLineEdit" name="firmware_path_le_5"/>
          </item>
          <item row="6" column="5">
           <widget class="QComboBox" name="firmware_crypto_cmb_5"/>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="firmware_selected_lb">
            <property name="text">
             <string>NO.</string>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QCheckBox" name="firmware_selected_chk_6">
            <property name="text">
             <string>#6</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="QLineEdit" name="firmware_path_le_0">
            <property name="minimumSize">
             <size>
              <width>150</width>
              <height>0</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="4" column="4">
           <widget class="QLineEdit" name="firmware_address_le_3"/>
          </item>
          <item row="4" column="5">
           <widget class="QComboBox" name="firmware_crypto_cmb_3"/>
          </item>
          <item row="1" column="3">
           <widget class="QToolButton" name="firmware_open_file_tb_0">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="3" column="4">
           <widget class="QLineEdit" name="firmware_address_le_2"/>
          </item>
          <item row="9" column="0">
           <widget class="QCheckBox" name="firmware_selected_chk_7">
            <property name="text">
             <string>#7</string>
            </property>
           </widget>
          </item>
          <item row="1" column="4">
           <widget class="QLineEdit" name="firmware_address_le_0">
            <property name="minimumSize">
             <size>
              <width>75</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="5">
           <widget class="QComboBox" name="firmware_crypto_cmb_0"/>
          </item>
          <item row="2" column="4">
           <widget class="QLineEdit" name="firmware_address_le_1"/>
          </item>
          <item row="4" column="0">
           <widget class="QCheckBox" name="firmware_selected_chk_3">
            <property name="text">
             <string>#3</string>
            </property>
           </widget>
          </item>
          <item row="3" column="3">
           <widget class="QToolButton" name="firmware_open_file_tb_2">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1" colspan="2">
           <widget class="QLineEdit" name="firmware_path_le_2"/>
          </item>
          <item row="5" column="0">
           <widget class="QCheckBox" name="firmware_selected_chk_4">
            <property name="text">
             <string>#4</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1" colspan="2">
           <widget class="QLineEdit" name="firmware_path_le_4"/>
          </item>
          <item row="7" column="1" colspan="2">
           <widget class="QLineEdit" name="firmware_path_le_6"/>
          </item>
          <item row="2" column="3">
           <widget class="QToolButton" name="firmware_open_file_tb_1">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="7" column="3">
           <widget class="QToolButton" name="firmware_open_file_tb_6">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1" colspan="2">
           <widget class="QLineEdit" name="firmware_path_le_7"/>
          </item>
          <item row="2" column="1" colspan="2">
           <widget class="QLineEdit" name="firmware_path_le_1"/>
          </item>
         </layout>
        </item>
        <item row="0" column="2">
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="6" column="0" colspan="3">
           <widget class="QPushButton" name="firmware_read_btn">
            <property name="text">
             <string>固件读取</string>
            </property>
           </widget>
          </item>
          <item row="5" column="2">
           <widget class="QLineEdit" name="firmware_read_size_le"/>
          </item>
          <item row="2" column="2">
           <widget class="QComboBox" name="firmware_transfer_protocol_cmb"/>
          </item>
          <item row="3" column="0" colspan="3">
           <widget class="QPushButton" name="firmware_update_btn">
            <property name="text">
             <string>固件升级</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="2">
           <widget class="QLabel" name="firmware_read_start_address_lb">
            <property name="text">
             <string>起始地址</string>
            </property>
           </widget>
          </item>
          <item row="4" column="2">
           <widget class="QLineEdit" name="firmware_read_start_address_le">
            <property name="minimumSize">
             <size>
              <width>75</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="5" column="0" colspan="2">
           <widget class="QLabel" name="firmware_read_size_lb">
            <property name="text">
             <string>大小</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="3">
           <widget class="QPushButton" name="firmware_combine_btn">
            <property name="text">
             <string>固件合并</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLineEdit" name="firmware_fill_byte_le">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QLabel" name="firmware_transfer_protocol_lb">
            <property name="layoutDirection">
             <enum>Qt::LayoutDirection::LeftToRight</enum>
            </property>
            <property name="text">
             <string>传输协议</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="firmware_fill_byte_lb">
            <property name="text">
             <string>固件填充</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="1">
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="serial_tab">
      <attribute name="title">
       <string>串口</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3" stretch="9,1">
       <item>
        <widget class="QPlainTextEdit" name="serial_rx_tx_pte">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPlainTextEdit" name="serial_tx_pte"/>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QCheckBox" name="serial_rx_tx_hex_chk">
             <property name="text">
              <string>HEX 显示</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="serial_tx_hex_chk">
             <property name="text">
              <string>HEX 发送</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="serial_clear_btn">
             <property name="text">
              <string>清空</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="serial_send_btn">
             <property name="text">
              <string>发送</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="log_tab">
      <property name="accessibleName">
       <string>a</string>
      </property>
      <attribute name="title">
       <string>日志</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QPlainTextEdit" name="log_pte">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>250</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections/>
</ui>
